tests/parametric/test_otel_interoperability.py

```diff
@@ -0,0 +1,518 @@
+        trace = find_trace_by_root(traces, OtelSpan(resource="my_new_resource2"))
+        assert len(trace) == 1
+
+    def test_span_links_basic(self, test_agent, test_library):
+        """
+            - Test that links set on a span created with the Datadog API are updated into the OTel API
+        """
+        with test_library:
+            with test_library.start_span("dd.span") as dd_span:
+                dd_span.add_link(
+                    parent_id=TEST_SPAN_ID,
+                    attributes=TEST_ATTRIBUTES,
+                    trace_id=TEST_TRACE_ID,
+                    tracestate=TEST_TRACESTATE,
+                )
+
+                otel_span = test_library.otel_current_span()
+                otel_span_links = otel_span.get_links()
+
+                assert len(otel_span_links) == 1
+
+                otel_link = otel_span_links[0]
+                assert otel_link["trace_id"] == TEST_TRACE_ID
+                assert otel_link["parent_id"] == TEST_SPAN_ID
+                assert otel_link["tracestate"] == TEST_TRACESTATE
+                assert otel_link["attributes"] == TEST_ATTRIBUTES
+
+                dd_span.finish()
+
+    def test_span_links_add(self, test_agent, test_library):
+        """
+            - Test that links set on a span created with the OTel API are updated into the Datadog API
+        """
+        with test_library:
+            with test_library.otel_start_span("otel.span") as otel_span:
+                current_span = test_library.current_span()
+
+                current_span.add_link(
+                    parent_id=TEST_SPAN_ID,
+                    attributes=TEST_ATTRIBUTES,
+                    trace_id=TEST_TRACE_ID,
+                    tracestate=TEST_TRACESTATE,
+                )
+
+                otel_span_links = otel_span.get_links()
+                assert len(otel_span_links) == 1
+
+                otel_link = otel_span_links[0]
+                assert otel_link["trace_id"] == TEST_TRACE_ID
+                assert otel_link["parent_id"] == TEST_SPAN_ID
+                assert otel_link["tracestate"] == TEST_TRACESTATE
+                assert otel_link["attributes"] == TEST_ATTRIBUTES
+
+                otel_span.end_span()
+
+    def test_concurrent_traces_in_order(self, test_agent, test_library):
+        """
+            - Basic concurrent traces and spans
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_root", span_kind=SK_SERVER) as otel_root:
+                with test_library.start_span(name="dd_child", parent_id=otel_root.span_id) as dd_child:
+                    with test_library.start_span(name="dd_root", parent_id=0) as dd_root:
+                        with test_library.otel_start_span(name="otel_child", parent_id=dd_root.span_id) as otel_child:
+                            otel_child.end_span()
+                        dd_root.finish()
+                otel_root.end_span()
```

Comment:
Is `end_span()` and `finish()` required as span should be ended with `with` block end?

Reply:
Since they are in order, this, indeed, shouldn't be required 😬  I believe there may be the same redundancy in some other test cases